flang/test/Lower/character-assignment.f90

```diff
@@ -78,7 +85,8 @@ subroutine assign_constant(lhs)
   character(*, 1) :: lhs
   ! CHECK: %[[lhs:.*]]:2 = fir.unboxchar %arg0
   ! CHECK: %[[cst:.*]] = fir.address_of(@{{.*}}) :
-  ! CHECK: %[[tmp]] = fir.alloca !fir.array<?x!fir.char<1>>, %{{.*}}
+  ! CHECK: %[[tmp_alloc:.*]] = fir.alloca !fir.char<1>, %{{.*}}
+  ! CHECK: %[[tmp:.*]] = fir.convert %[[tmp_alloc]] : (!fir.ref<!fir.char<1>>) -> !fir.ref<!fir.char<1,?>>
```

Comment:
Can't this be
```mlir
fir.alloca !fir.char<1, ?>, ...
```
and get rid of the `fir.convert`?

Reply:
It actually can, thanks. I have updated `createCharacterTemp` to do that.